parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetRecordWriter.java

```diff
@@ -68,7 +67,7 @@ public ParquetRecordWriter(
       MessageType schema,
       Map<String, String> extraMetaData,
       int blockSize, int pageSize,
-      BytesCompressor compressor,
+      BytesInputCompressor compressor,
```

Comment:
Semantic versioning check failed on these removed constructors. Though `BytesCompressor` is marked as deprecated, I think you still should use it here instead of `BytesInputCompressor`, so that this PR can be reased in a 1.x release.

Reply:
Ah! Thanks for pointing that out, @nandorKollar . Apparently, this and the other constructor taking BytesInputCompressor isn't used (at least I didn't find any references to it within the parquet project). I wonder if it would be ok with semantic versioning check to get rid of them. Going to try that.

Reply:
Turns out removing constructors even though they are not called anywhere in the code isn't allowed either. Switching the type back to BytesCompressor did the trick.